### DISCUSSION TITLE
Allow skin component colors to match the current difficulty star rating

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#31265] [Ideas]
As the title says, I think it would be cool to have some sort of checkbox to allow skin elements (like boxes and text) to automatically change their color to match that of the currently selected difficulty (aka the star rating color).

Maybe even a dropdown to choose between the actual difficulty color and the *readable one* above it (from the star rating display spectrum):
![image](https://github.com/user-attachments/assets/b3f7b7f6-f4a8-47b1-954a-046b826e27c3)

Here's a simple mockup I made of the new settings using Figma (the wording is very random, just to fill in the blanks).

![image](https://github.com/user-attachments/assets/5b6db936-d31b-42ea-a2c5-b0cb97b6aef9)

I think the star symbol in the star rating is a different color, but I guess it's no big deal to just... ignore that detail. Not sure.
Anyway, here are some (updated) examples in action:

![ex 1](https://github.com/user-attachments/assets/e6631474-279b-48d2-bafa-eaece25d5a1c)

![ex 2](https://github.com/user-attachments/assets/a6ac9eb4-1e09-4fd9-b007-dcd1edac44eb)

**Comment by peppy** [comment]
I like this proposal, but as with the constantly increasing amount of common skin settings, I'd hope this is done in a way that can be reused across every skin element with colour support. Some kind of composition pattern.
